src/Host.cpp

```diff
@@ -2598,3 +2628,956 @@ void Host::setCompactInputLine(const bool state)
+    }
+
+    //Dont create Userwindow if there is a Label with the same name already. It breaks the UserWindow
+    auto pL = mpConsole->mLabelMap.value(name);
+    if (pL) {
+        return {false, QStringLiteral("label with the name \"%1\" exists already. userwindow name has to be unique").arg(name)};
+    }
+
+    auto hostName(getName());
+    auto console = mpConsole->mSubConsoleMap.value(name);
+    auto dockwidget = mpConsole->mDockWidgetMap.value(name);
+
+    if (!console && !dockwidget) {
+        // The name is not used in either the QMaps of all user created TConsole
+        // or TDockWidget instances - so we can make a NEW one:
+        dockwidget = new TDockWidget(this, name);
+        dockwidget->setObjectName(QStringLiteral("dockWindow_%1_%2").arg(hostName, name));
+        dockwidget->setContentsMargins(0, 0, 0, 0);
+        dockwidget->setFeatures(QDockWidget::AllDockWidgetFeatures);
+        dockwidget->setWindowTitle(name);
+        mpConsole->mDockWidgetMap.insert(name, dockwidget);
+        // It wasn't obvious but the parent passed to the TConsole constructor
+        // is sliced down to a QWidget and is NOT a TDockWidget pointer:
+        console = new TConsole(this, TConsole::UserWindow, dockwidget->widget());
+        console->setObjectName(QStringLiteral("dockWindowConsole_%1_%2").arg(hostName, name));
+        // Without this the TConsole instance inside the TDockWidget will be
+        // left being called the default value of "main":
+        console->mConsoleName = name;
+        console->setContentsMargins(0, 0, 0, 0);
+        dockwidget->setTConsole(console);
+        console->layerCommandLine->hide();
+        console->mpScrollBar->hide();
+        mpConsole->mSubConsoleMap.insert(name, console);
+        dockwidget->setStyleSheet(mProfileStyleSheet);
+        mudlet::self()->addDockWidget(Qt::RightDockWidgetArea, dockwidget);
+        console->setFontSize(10);
+    }
+    if (!console || !dockwidget) {
+        return {false, QStringLiteral("cannot create userwindow \"%1\": name collision").arg(name)};
+    }
+
+    // The name is used in BOTH the QMaps of all user created TConsole
+    // and TDockWidget instances - so we HAVE an existing user window,
+    // Lets confirm this:
+    Q_ASSERT_X(console->getType() == TConsole::UserWindow, "mudlet::openWindow(...)", "An existing TConsole was expected to be marked as a User Window type but it isn't");
+    dockwidget->update();
+
+    if (loadLayout && !dockwidget->hasLayoutAlready) {
+        mudlet::self()->loadWindowLayout();
+        dockwidget->hasLayoutAlready = true;
+    }
+    dockwidget->show();
+
+    if (!autoDock) {
+        dockwidget->setAllowedAreas(Qt::NoDockWidgetArea);
+    } else {
+        dockwidget->setAllowedAreas(Qt::AllDockWidgetAreas);
+    }
+
+    if (area.isEmpty()) {
+        return {true, QString()};
+    }
+
+    if (area == QLatin1String("f") || area == QLatin1String("floating")) {
```

Comment:
> <off-topic>I am sure I have mentioned this before but

If you already know this, why mention here and not open or append to a seperate issue? It will not be solved here..!

Reply:
Because seeing it reminded me of it and I wasn't minded to (due to a low ☕ state) go off and hunt for a possibly non-existing issue or file a new issue when I had a whole lot more of this PR to look through. I guess there could have been a bit of unconscious [squeaky-wheelism](https://en.wikipedia.org/wiki/The_squeaky_wheel_gets_the_grease)...